source/patterns/@aws-solutions-constructs/aws-fargate-sns/lib/index.ts

```diff
@@ -0,0 +1,162 @@
+   *
+   * @default - none
+   */
+  readonly publicApi: boolean;
+  /**
+   * Optional properties to create a new ECS cluster
+   */
+  readonly clusterProps?: ecs.ClusterProps;
+  /**
+   * The arn of an ECR Repository containing the image to use
+   * to generate the containers
+   *
+   * format:
+   *   arn:aws:ecr:[region]:[account number]:repository/[Repository Name]
+   */
+  readonly ecrRepositoryArn?: string;
+  /**
+   * The version of the image to use from the repository
+   *
+   * @default - 'latest'
+   */
+  readonly ecrImageVersion?: string;
+  readonly existingImageObject?: ecs.ContainerImage;
+  readonly containerDefinitionProps?: ecs.ContainerDefinitionProps | any;
+  readonly fargateTaskDefinitionProps?: ecs.FargateTaskDefinitionProps | any;
+  /**
+   * Optional properties to override default values for the Fargate service.
+   * Service will set up in the Public or Isolated subnets of the VPC by default,
+   * override that (e.g. - choose Private subnets) by setting vpcSubnets on this
+   * object.
+   */
+  readonly fargateServiceProps?: ecs.FargateServiceProps | any;
+  /**
+   * A Fargate Service already instantiated (probably by another Solutions Construct). If
+   * this is specified, then no props defining a new service can be provided, including:
+   * existingImageObject, ecrImageVersion, containerDefintionProps, fargateTaskDefinitionProps,
+   * ecrRepositoryArn, fargateServiceProps, clusterProps, existingClusterInterface
+   *
+   * @default - none
+   */
+  readonly existingFargateServiceObject?: ecs.FargateService;
+
+  /**
+   * Existing instance of SNS Topic object, providing both this and topicProps will cause an error..
+   *
+   * @default - Default props are used
+   */
+  readonly existingTopicObject?: sns.Topic;
+  /**
+   * Optional user provided properties to override the default properties for the SNS topic.
+   *
+   * @default - Default properties are used.
+   */
+  readonly topicProps?: sns.TopicProps;
+  /**
+   * Optional Name for the SNS topic arn environment variable set for the Lambda function.
+   *
+   * @default - None
+   */
+  readonly topicArnEnvironmentVariableName?: string;
+  /**
+   * Optional Name for the SNS topic name environment variable set for the Lambda function.
+   *
+   * @default - None
+   */
+  readonly topicNameEnvironmentVariableName?: string;
+
+  readonly existingContainerDefinitionObject?: ecs.ContainerDefinition;
+}
+
+export class FargateToSns extends Construct {
+  public readonly snsTopic: sns.Topic;
+  public readonly service: ecs.FargateService;
+  public readonly vpc: ec2.IVpc;
+  public readonly container: ecs.ContainerDefinition;
+
+  constructor(scope: Construct, id: string, props: FargateToSnsProps) {
+    super(scope, id);
+    defaults.CheckProps(props);
+    defaults.CheckFargateProps(props);
+
+    if (props.existingVpc) {
+      this.vpc = props.existingVpc;
+    } else {
+      this.vpc = defaults.buildVpc(scope, {
```

Comment:
Can we create ObtainVpc? Or didn't we learn that buildVpc already does what we want and we don't need all this?